easybuild/easyconfigs/h/hypothesis/hypothesis-4.44.2-GCCcore-8.3.0-Python-3.7.4.eb

```diff
@@ -0,0 +1,33 @@
+# This easyconfig was created by Simon Branford of the BEAR Software team at the University of Birmingham.
+easyblock = 'PythonBundle'
+
+name = 'hypothesis'
+version = '4.44.2'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = "https://github.com/HypothesisWorks/hypothesis"
+description = """Hypothesis is an advanced testing library for Python. It lets you write tests which are parametrized
+ by a source of examples, and then generates simple and comprehensible examples that make your tests fail. This lets
+ you find more bugs in your code with less work."""
+
+toolchain = {'name': 'GCCcore', 'version': '8.3.0'}
+
+builddependencies = [('binutils', '2.32')]
+
+dependencies = [('Python', '3.7.4')]
```

Comment:
@branfosj hypothesis still works with Python 2, no?
in that case please use `multi_deps` as was done for `hypothesis-4.23.4-GCCcore-8.2.0.eb`?
(see also the email of Kenneth to the mailing list from 1 day ago)

Reply:
I've just tried to `multi_deps` it and I'm hitting an odd error: https://gist.github.com/branfosj/61a8bf5135f2cc422c1f8d545b5da7f3

https://gist.github.com/branfosj/61a8bf5135f2cc422c1f8d545b5da7f3#file-easybuild-hypothesis-4-44-2-20191116-190723-sallh-log-L1124 shows: `TypeError: attrs() got an unexpected keyword argument 'eq'` which is unexpected. This feature got added to `attrs` in 19.2 but I'm using 19.3 here, which suggests that we're picking up the `attrs` 19.1 that is included in the Python 2.7.16 and 3.7.4 installs instead. However, this doesn't happen when I build the Python 3.7.4 specific hypothesis module.

And,
```
diff -r ../../../../../../EL7/EL7-cascadelake/software/hypothesis/4.44.2-GCCcore-8.3.0-Python-3.7.4/lib/python3.7/site-packages/ ../../../../../../EL7/EL7-cascadelake/software/hypothesis/4.44.2-GCCcore-8.3.0/lib/python3.7/site-packages/
```
shows that the two `lib/python3.7/site-packages` directories are identical.

I'm running at version 4.0.1 for framework, easyblocks, and easyconfigs.

Reply:
You can't easily override a Python package from the `Python` installation when using `multi_deps`, we've run into that earlier this week for something else, basically because for `multi_deps` to work we don't use `$PYTHONPATH` but [site.addsitedir](https://docs.python.org/3/library/site.html#site.addsitedir), and apparently the latter only append, there is no way to prepend (see also https://bugs.python.org/issue7744).

So I guess we'll have to go forward with Python 3-only here.

Is this a dependency for something else?

Reply:
I've started work on PyTorch 1.3.1 and this is for that. The pull requests I have done today are all in support of that.

Reply:
We're stuck to a single `Python` version there anyway, due to `SciPy-bundle`, so OK.